pkg/config/config.go

```diff
@@ -128,9 +128,45 @@ func NewConfig(configPath string) (*Config, error) { // nolint // function will
 		}
 	}
 
+	err = cfg.Validate()
+	if err != nil {
+		return nil, errors.Wrap(err, "config validation failed")
+	}
+
 	return cfg, nil
 }
 
+func (cfg *Config) Validate() error {
+	if cfg.SCIM.UserObjectType == "" {
+		return errors.New("scim.user_object_type is required")
+	}
+	if cfg.SCIM.IdentityObjectType == "" {
+		return errors.New("scim.identity_object_type is required")
+	}
+	if cfg.SCIM.IdentityRelation == "" {
+		return errors.New("scim.identity_relation is required")
+	} else {
+		object, relation, found := strings.Cut(cfg.SCIM.IdentityRelation, "#")
```

Comment:
We can store `object` and `relation` so we don't have to parse them again on every call.